src/parser/parser_support.c

```diff
@@ -11262,7 +11431,10 @@ pt_convert_dblink_synonym (PARSER_CONTEXT * parser, PT_NODE * spec, void *has_sy
 	      spec->info.spec.range_var = parser_new_node (parser, PT_NAME);
 	      spec->info.spec.range_var->info.name.original = pt_append_string (parser, NULL, synonym_name);
 	    }
-	  *(bool *) has_synonym = true;
+	  if (has_synonym)
```

Comment:
Why do we still need to use has_synonym?
It is only the pt_convert_dblink_insert_query function that passes has_synonym when calling pt_convert_dblink_synonym.

The reason is to call parser_free_node function for 'node->info.insert.spec->info.spec.range_var' added in pt_convert_dblink_synonym.
Is it correct to call the parser_free_node function when has_synonym is true?

Reply:
Ok, I aggree with you. I will fix it.